enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/CausalClusteringSettings.java

```diff
@@ -121,14 +127,37 @@
             setting( "causal_clustering.initial_discovery_members", list( ",", ADVERTISED_SOCKET_ADDRESS ),
                     NO_DEFAULT );
 
+    public enum DiscoveryType
+    {
+        DNS,
+        LIST
+    }
+
+    @Description( "Configure the discovery type" )
+    public static final Setting<DiscoveryType> discovery_type =
+            setting( "causal_clustering.discovery_type", options( DiscoveryType.class ), DiscoveryType.LIST.name() );
+
+    public static ResolutionResolver chooseResolver( Config config, LogProvider logProvider,
```

Comment:
Would it perhaps make sense to move this somewhere else outside the Settings-class? I think it might get a bit polluted here otherwise.